### DISCUSSION TITLE
Access LinearTransform among many inputs

### DISCUSSION THREAD

**Discussion by ReiniertB** [lassoan/pyigtl#9] [Q&A]
Hi,

I was able to connect a BrainLab system to 3D slicer with the help of pyigtl, which is very helpful.
However, when connecting, there are many inputs as you can see in the image below.
I want to access the LinearTransform from the Pointer as a matrix in my python code but have not been successful so far.
How should I handle this? Should I add for example a server connector in Slicer that can send this to my code?
Should that have the same name and access port as the BrainLab system?

Or is it possible to access this a different way? Thanks in advance!

![image](https://user-images.githubusercontent.com/75205068/191694210-4b702c24-6f52-45e5-9021-dde1514741bb.png)

**Comment by lassoan** [comment]
You can run all your Python scripts within 3D Slicer. There is no need for pyigtl, as 3D Slicer has a built-in OpenIGTLink server and client. You can access the matrix like this:

```python
pointerTransformNode =slicer.util.getNode('Pointer')
print(slicer.util.arrayFromTransformMatrix(pointerTransformNode))
```
